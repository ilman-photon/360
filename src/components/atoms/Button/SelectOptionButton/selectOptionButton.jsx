import * as React from "react";
import FormControl from "@mui/material/FormControl";
import { InputLabel, MenuItem, Select } from "@mui/material";
import { ThemeProvider, styled, alpha } from "@mui/material/styles";

export const CustomFormControl = styled((props) => <FormControl {...props} />)(
  ({ theme }) => ({
    "& .MuiFilledInput-root": {
      border: "1px solid #e2e2e1",
      overflow: "hidden",
      borderRadius: 4,
      height: 52,
      backgroundColor: theme.palette.mode === "light" ? "#fcfcfb" : "#2b2b2b",
      transition: "none",
      "&:hover": {
        backgroundColor: "transparent",
      },
      "&.Mui-error": {
        borderColor: "#FF0000",
        backgroundColor: "#FF000010",
      },
      "&.Mui-focused": {
        backgroundColor: "transparent",
        boxShadow: `${alpha(theme.palette.primary.main, 0.25)} 0 0 0 2px`,
        borderColor: theme.palette.primary.main,
      },
    },
  })
);

export const CustomSelect = styled((props) => <Select {...props} />)(
  ({ theme }) => ({
    "& .MuiFilledInput-root": {
      border: "1px solid #e2e2e1",
      overflow: "hidden",
      borderRadius: 4,
      height: 52,
      backgroundColor: theme.palette.mode === "light" ? "#fcfcfb" : "#2b2b2b",
      transition: "none",
      "&:hover": {
        backgroundColor: "transparent",
        borderBottom: "0px",
      },
      "&:before": {
        borderColor: "transparent !important",
        borderBottom: "0px",
      },
      "&:after": {
        borderColor: "transparent !important",
        borderBottom: "0px",
      },
      "&.Mui-error": {
        borderColor: "#FF0000",
        backgroundColor: "#FF000010",
      },
      "&.Mui-focused": {
        backgroundColor: "transparent",
        borderColor: "0px",
      },
    },
  })
);

export default function SelectOptionButton({
  label = "",
  options = [],
  labelId,
  id,
  ...props
}) {
  const [optionValue, setOptionValue] = React.useState("");

  const handleChange = (event) => {
    setOptionValue(event.target.value);
  };

  return (
    <CustomFormControl variant="filled" {...props} style={{ width: "100%" }}>
      <InputLabel id="demo-simple-select-label">{label}</InputLabel>
      <Select
        labelId={labelId ? labelId : null}
        id={id ? id : null}
        value={optionValue}
        label={label}
        onChange={handleChange}
      >
        {options.map((option, idx) => {
          return (
            <MenuItem key={idx} value={option}>
              {option}
            </MenuItem>
          );
        })}
      </Select>
    </CustomFormControl>
  );
}
