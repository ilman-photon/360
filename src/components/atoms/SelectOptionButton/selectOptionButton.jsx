import * as React from "react";
import FormControl from "@mui/material/FormControl";
import { FormHelperText, InputLabel, MenuItem, Select } from "@mui/material";
import { styled, alpha } from "@mui/material/styles";
import { colors } from "../../../styles/theme";
import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";

export const CustomFormControl = styled((props) => <FormControl {...props} />)(
  ({ theme }) => ({
    "& .MuiFilledInput-root": {
      border: "1px solid #e2e2e1",
      overflow: "hidden",
      borderRadius: 4,
      height: 52,
      backgroundColor: theme.palette.mode === "light" ? "#fcfcfb" : "#2b2b2b",
      transition: "none",
      "&:hover": {
        backgroundColor: "transparent",
      },
      "&:before": {
        borderColor: "transparent !important",
        borderBottom: "0px !important",
        transition: "none",
      },
      "&:focus": {
        backgroundColor: "#fff",
      },
      "&:after": {
        borderColor: "transparent !important",
        borderBottom: "0px !important",
        transition: "none",
      },
      "&.Mui-error": {
        borderColor: "#FF0000",
        backgroundColor: "#FF000010",
      },
      "&.Mui-focused": {
        backgroundColor: "transparent",
        boxShadow: `${alpha(theme.palette.primary.main, 0.25)} 0 0 0 2px`,
        borderColor: "transparent",
      },
    },
  })
);

export default function SelectOptionButton({
  label = "",
  options = [],
  labelId,
  id,
  value = "",
  onChange,
  renderMenuListUI = null,
  ariaLabel = "",
  helperText = "",
  inputProps = {},
  renderValue,
  ...props
}) {
  return (
    <CustomFormControl
      variant="filled"
      sx={{ ...props.sx }}
      style={{ width: "100%" }}
    >
      <InputLabel
        aria-label={ariaLabel}
        role={props.role}
        id="demo-simple-select-label"
        sx={{
          fontSize: "16px",
          "&.MuiInputLabel-shrink": {
            color: colors.darkGreen,
            fontWeight: 600,
            fontFamily: "Museo Sans",
            fontSize: "1rem",
          },
          "&.Mui-error": {
            borderColor: "#B93632",
            backgroundColor: "#FBF4F4",
            color: "#6C6C6C",
          },
          ...props.inputPropsSx,
        }}
        {...props}
      >
        {label}
      </InputLabel>
      <Select
        labelId={labelId ? labelId : null}
        id={id ? id : null}
        value={value}
        label={label}
        onChange={onChange}
        IconComponent={KeyboardArrowDownIcon}
        sx={{
          fontSize: "16px",
          "& .MuiInputBase-input:focus": {
            backgroundColor: "#fff",
          },
        }}
        inputProps={{
          "aria-label": `${label}. ${value}`,
          "aria-live": "polite",
          "data-testid": `select-${id}`,
          ...inputProps,
        }}
        MenuProps={props.MenuProps}
        {...props.menuProps}
        renderValue={
          renderValue ||
          ((select) => {
            return select;
          })
        }
        {...props}
      >
        {options.map((option, idx) => {
          if (renderMenuListUI) {
            return renderMenuListUI(option, idx);
          } else {
            return (
              <MenuItem
                key={idx}
                value={option}
                sx={{ fontSize: "16px", whiteSpace: "pre-wrap" }}
                // inputProps={{
                //   "aria-label": `${option}`,
                //   "aria-live": "polite",
                // }}
              >
                {option}
              </MenuItem>
            );
          }
        })}
      </Select>
      <FormHelperText
        sx={{
          color: props.error ? colors.error : colors.darkGreen,
          fontSize: "12px",
        }}
      >
        {helperText}
      </FormHelperText>
    </CustomFormControl>
  );
}
