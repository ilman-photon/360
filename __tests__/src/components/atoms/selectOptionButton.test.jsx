import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import SelectOptionButton from "../../../../src/components/atoms/SelectOptionButton/selectOptionButton";

describe("SelectOptioButton Components", () => {
  let container;
  beforeEach(() => {
    container = render(<SelectOptionButton option={[
        "What was the first concert you attended?",
        "In what city or town did your parents meet?",
        "What was the make and model of your first car?",
        "Who is your all-time favorite movie character?",
        "What was your favorite cartoon character during your childhood?",
        "What was the first book you read?",
        "What was the first thing you learned to cook?",
        "What was the first film you saw in a theater?",
        "Where did you go the first time you flew on a plane?",
        "What is your favorite cold-weather activity?",
      ]}
      label={`Question`}
      labelId={`question-label`}
      id={`question-id`}
      />);
  });
  it("select option render", () => {
    expect(container).toMatchSnapshot();
  });

});
