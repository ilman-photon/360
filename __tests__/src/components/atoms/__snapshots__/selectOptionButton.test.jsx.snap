// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectOptioButton Components select option render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root css-1nn82kk-MuiFormControl-root"
        option="What was the first concert you attended?,In what city or town did your parents meet?,What was the make and model of your first car?,Who is your all-time favorite movie character?,What was your favorite cartoon character during your childhood?,What was the first book you read?,What was the first thing you learned to cook?,What was the first film you saw in a theater?,Where did you go the first time you flew on a plane?,What is your favorite cold-weather activity?"
        style="width: 100%;"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-18ljltl-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          id="demo-simple-select-label"
        >
          Question
        </label>
        <div
          class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-baqyug-MuiInputBase-root-MuiFilledInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="question-label question-id"
            class="MuiSelect-select MuiSelect-filled MuiFilledInput-input MuiInputBase-input css-d9oaum-MuiSelect-select-MuiInputBase-input-MuiFilledInput-input"
            id="question-id"
            role="button"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconFilled css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root css-1nn82kk-MuiFormControl-root"
      option="What was the first concert you attended?,In what city or town did your parents meet?,What was the make and model of your first car?,Who is your all-time favorite movie character?,What was your favorite cartoon character during your childhood?,What was the first book you read?,What was the first thing you learned to cook?,What was the first film you saw in a theater?,Where did you go the first time you flew on a plane?,What is your favorite cold-weather activity?"
      style="width: 100%;"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-18ljltl-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        id="demo-simple-select-label"
      >
        Question
      </label>
      <div
        class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-baqyug-MuiInputBase-root-MuiFilledInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="question-label question-id"
          class="MuiSelect-select MuiSelect-filled MuiFilledInput-input MuiInputBase-input css-d9oaum-MuiSelect-select-MuiInputBase-input-MuiFilledInput-input"
          id="question-id"
          role="button"
          tabindex="0"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconFilled css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
